org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/LookupEnvironment.java

```diff
@@ -1189,6 +1181,28 @@ public PlainPackageBinding createPlainPackage(char[][] compoundName) {
 	return packageBinding.getIncarnation(this.module);
 }
 
+private NameEnvironmentAnswer findType(char[] typeName, char[][] packageName) {
+	if (this.nameEnvironment instanceof IReleaseAwareNameEnvironment releaseAware) {
+		return releaseAware.findType(typeName, packageName, getRelease());
+	}
+	return this.nameEnvironment.findType(typeName, packageName);
+}
+
+private NameEnvironmentAnswer findType(char[][] typeName) {
+	if (this.nameEnvironment instanceof IReleaseAwareNameEnvironment releaseAware) {
+		return releaseAware.findType(typeName, getRelease());
+	}
+	return this.nameEnvironment.findType(typeName);
+}
+
+private int getRelease() {
+	int release =  (int) Double.parseDouble(CompilerOptions.versionFromJdkLevel(this.globalOptions.targetJDK));
```

Comment:
Apparently you didn't find any utils for encoding JDK versions as (human readable) int. Given that nobody needed such encoding until now, I would prefer to keep with one of the existing encodings, either `ClassFileConstants.JDK9` ... or `ClassFileConstants.MAJOR_VERSION_9` ...,  where the latter format can be obtained like this `(int) (this.globalOptions.targetJDK >> 16)`.

Reply:
I now replaced the parsing with the suggested simple form. One needs a "human readable" int because that is the actual value used to lookup the correct multi-release folder (and the MR-spec is all talking about that value). Encoding it as something else would make this not only harder to debug but also requires converting the value to the "real" one on multiple places.